src/simulator.zig

```diff
@@ -30,10 +30,17 @@ pub fn main() !void {
     var args = std.process.args();
 
     // Skip argv[0] which is the name of this executable:
-    _ = args.nextPosix();
+    _ = args_next(&args, allocator);
 
     const seed_random = std.crypto.random.int(u64);
-    const seed = if (args.nextPosix()) |bytes| parse_seed(bytes) else seed_random;
+    const seed = seed_from_arg: {
+        const arg_two = args_next(&args, allocator);
+        if (arg_two == null) break :seed_from_arg seed_random;
+
+        const parsed_arg_seed = parse_seed(arg_two.?);
+        defer allocator.free(arg_two.?);
+        break :seed_from_arg parsed_arg_seed;
+    };
```

Comment:
This can be cleaned up by using `orelse` and getting rid of an uneeded variable.
```suggestion
    const seed = seed_from_arg: {
        const arg_two = args_next(&args, allocator) orelse break :seed_from_arg seed_random;
        defer allocator.free(arg_two);
        break :seed_from_arg parse_seed(arg_two);
    };
```
Since this block is quite short, using a shorter label like the commonly used `blk:` might be more readable, though that's quite subjective.